pkg/config/config_new.go

```diff
@@ -0,0 +1,161 @@
+// config/config.go
+package config
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/cloudposse/atmos/pkg/schema"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var DefaultConfigHandler = New()
+
+// ConfigHandler holds the application's configuration.
+type ConfigHandler struct {
+	atmosConfig *schema.AtmosConfiguration
+	v           *viper.Viper
+}
+
+// ConfigOptions defines options for adding a configuration parameter.
+type ConfigOptions struct {
+	FlagName     string      // Custom flag name (optional)
+	EnvVar       string      // Custom environment variable (optional)
+	Description  string      // Flag description
+	Key          string      // Key of the data in atmosConfiguration
+	DefaultValue interface{} // Default value of the data in atmosConfiguration
+}
+
+// New creates a new Config instance with initialized Viper.
+func New() *ConfigHandler {
+	v := viper.New()
+	v.SetConfigType("yaml")
+	v.SetTypeByDefaultValue(true)
+	configHandler := &ConfigHandler{
+		v:           v,
+		atmosConfig: &schema.AtmosConfiguration{},
+	}
+	return configHandler
+}
+
+// AddConfig adds a configuration parameter to both Cobra and Viper with options.
+func (c *ConfigHandler) AddConfig(cmd *cobra.Command, opts *ConfigOptions) {
+	key := opts.Key
+	defaultValue := opts.DefaultValue
+	// Set default value in Viper
+	c.v.SetDefault(key, defaultValue)
+
+	// Determine flag name
+	flagName := opts.FlagName
+	if flagName == "" {
+		flagName = strings.ReplaceAll(key, ".", "-")
+	}
+
+	// Register flag with Cobra
+	flagSet := cmd.PersistentFlags()
+
+	switch v := defaultValue.(type) {
+	case string:
+		flagSet.String(flagName, v, opts.Description)
+	case int:
+		flagSet.Int(flagName, v, opts.Description)
+	case bool:
+		flagSet.Bool(flagName, v, opts.Description)
+	case []string:
+		flagSet.StringSlice(flagName, v, opts.Description)
+	default:
+		panic(fmt.Sprintf("unsupported type for key %s", key))
+	}
+
+	// Bind the flag to Viper
+	if err := c.v.BindPFlag(key, flagSet.Lookup(flagName)); err != nil {
+		panic(fmt.Sprintf("failed to bind %s: %v", key, err))
+	}
+
+	// Handle environment variable binding
+	if opts.EnvVar != "" {
+		c.BindEnv(key, opts.EnvVar)
+	}
+}
+
+// load reads and merges the configuration.
+func (c *ConfigHandler) load(configAndStacksInfo *schema.ConfigAndStacksInfo) error {
+	setDefaultConfiguration(c.v)
+	setEnv(c.v)
+	// Read config file if exists (non-blocking)
+	if err := loadConfigSources(c.v, configAndStacksInfo.AtmosCliConfigPath); err != nil {
+		return err
+	}
+	if len(configAndStacksInfo.AtmosConfigFilesFromArg) > 0 || len(configAndStacksInfo.AtmosConfigDirsFromArg) > 0 {
+		err := loadConfigFromCLIArgs(c.v, configAndStacksInfo, c.atmosConfig)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
```

Comment:
--config,--config-path flag allows you to specify configuration files and Only the configuration files specified by this flag will be loaded
```suggestion
	if len(configAndStacksInfo.AtmosConfigFilesFromArg) > 0 || len(configAndStacksInfo.AtmosConfigDirsFromArg) > 0 {
		err := loadConfigFromCLIArgs(c.v, configAndStacksInfo, c.atmosConfig)
		if err != nil {
			return err
		}
		return nil
	}
	// Read config file if exists (non-blocking)
	if err := loadConfigSources(c.v, configAndStacksInfo.AtmosCliConfigPath); err != nil {
		return err
	}

```

Reply:
Done